clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -263,7 +263,14 @@ public void run() {
         while (!forceClose && transactionManager != null && transactionManager.hasOngoingTransaction()) {
             if (!transactionManager.isCompleting()) {
                 log.info("Aborting incomplete transaction due to shutdown");
-                transactionManager.beginAbort();
+
+                try {
+                    // It is possible for the transaction manager to throw errors when aborting. Catch these
```

Comment:
Are these errors newly added in the PR?

Reply:
No, this was discovered when auditing the `TransactionManager` call sites.

Reply:
Ah more bug fixes then :) thanks!